scripts/questions_gen.py

```diff
@@ -0,0 +1,305 @@
+    print("*** start evaluation")
+    faithfulness = FaithfulnessEvaluator(service_context=service_context)
+    relevancy = RelevancyEvaluator(service_context=service_context)
+    correctness = CorrectnessEvaluator(service_context=service_context,score_threshold=2.0 ,parser_function=eval_parser)
+
+    runner = BatchEvalRunner(
+                                        { "faithfulness": faithfulness, "relevancy": relevancy ,
+                                        },
+                                            workers=100, show_progress=True
+                                    )
+        
+    eval_results = await runner.aevaluate_queries( index.as_query_engine(similarity_top_k=SIMILARITY, \
+                                                                                service_context=service_context), \
+                                                                            queries=eval_questions ) 
+        
+    results["faithfulness"] = get_eval_breakdown("faithfulness", eval_results)
+    results["relevancy"] = get_eval_breakdown("relevancy", eval_results) 
+            
+    end_time = time.time()
+    execution_time_seconds = end_time - start_time    
+    print(f"** completed faithfulness,relevancy evaluation: execution time in min: {execution_time_seconds/60}")
+        
+            # correctness
+        
+    engine = index.as_query_engine(similarity_top_k=SIMILARITY, service_context=service_context)
+    res_table = []
+            
+    for query in eval_questions: 
+        res_row ={}
+        summary = engine.query(query)
+        referenced_documents = "\n".join(
+                    [
+                        source_node.node.metadata["file_name"]
+                        for source_node in summary.source_nodes
+                    ]
+                )
+                
+        result =correctness.evaluate(
+                    query=query,
+                    response=summary.response,
+                    reference=summary.source_nodes[0].text,
+                    )
+                
+        res_row["query"] = query
+        res_row["response"] = summary.response
+        res_row["ref"] = referenced_documents 
+        res_row["ref_doc"] = summary.source_nodes[0].text
+        res_row["ref_doc_score"] = summary.source_nodes[0].score
+        res_row["passing"] = result.passing
+        res_row["feedback"] = result.feedback
+        res_row["score"] = result.score
+        total_correctness_score.append(float(result.score))
+        res_table.append(res_row)
+        results["correctness"] = res_table
+                
+        end_time = time.time()
+        execution_time_seconds = end_time - start_time
+        print(f"*** Completed correctness evaluation: execution time in min: {execution_time_seconds/60}")
+        full_results.append(results)
+    
+
+
+async def main(): 
```

Comment:
why we run it asynchronously? it's one process with one coroutine, right

Reply:
llama_index evel runs async.